xbmc/threads/Condition.h

```diff
@@ -20,14 +20,59 @@
 
 #pragma once
 
-#include "threads/platform/Condition.h"
-
+#include "threads/SingleLock.h"
+#include "threads/Helpers.h"
 #include "threads/SystemClock.h"
-#include <stdio.h>
+
+#include <condition_variable>
+#include <chrono>
 
 namespace XbmcThreads
 {
 
+  /**
+   * This is a thin wrapper around boost::condition_variable. It is subject
+   *  to "spurious returns" as it is built on boost which is built on posix
+   *  on many of our platforms.
+   */
+  class ConditionVariable : public NonCopyable
```

Comment:
Do we want to use `NonCopyable` in new code? Wouldn't it be better to use `= delete`?

Reply:
@pkerling I can fix this. This isn't new code though. The class was just moved and modified from a platform specific version to make it non-platform specific.

Reply:
Ah OK, didn't check where this was from. Still, can as well fix it while we're at it :)